### PR TITLE
Additional transforms

### PR DESCRIPTION
**Pull request by mibaumgartner**
## Short Description
* RenameKeys Transform
* optional dtpye cast to onehot encoding
* OneOf selection trafo
* ArgMax Trafo
* move device functional to tensor functional file (breaking change)

### PR Implementer
This is a small checklist for the implementation details of this PR.
If you submit a PR, please look at these points (don't worry about the `RisingTeam`
and `Reviewer` workflows, the only purpose of those is to have a compact view of
the steps)

If there are any questions regarding code style or other conventions check out our 
[summary](https://github.com/PhoenixDL/rising/blob/master/CONTRIBUTING.md).

- [x] Implementation
- [x] Docstrings & Typing
- [x] Check `__all__` sections and `__init__`
- [x] Unittests (look at the line coverage of your tests, the goal is 100%!)
- [x] Update notebooks & documentation if necessary
- [x] Pass all tests
- [] Add the checksum of the last implementation commit to the Changelog

### RisingTeam
<details>
  <summary>RisingTeam workflow</summary>

  - [ ] Add pull request to project (optionally delete corresponding project note)
  - [ ] Assign correct label (if you don't have permission to do this, someone will do it for you. 
      Please make sure to communicate the current status of the pr.)
  - [ ] Does this PR close an Issue? (add `closes #IssueNumber` at the bottom if 
        not already in description)
</details>

### Reviewer
<details>
  <summary>Reviewer workflow</summary>
  
  - [ ] Do all tests pass? (Unittests, NotebookTests, Documentation)
  - [ ] Does the implementation follow `rising` design conventions?
  - [ ] Are the tests useful? (!!!) Are additional tests needed? 
        Can you think of critical points which should be covered in an additional test?
  - [ ] Optional: Check coverage locally / Check tests locally if GPU is necessary to execute
</details>

### CODE REVIEW

File: rising/transforms/channel.py [hunk 1]
```diff
@@ -1,10 +1,11 @@
-from typing import Sequence
+import torch
+from typing import Sequence, Optional
```
Comment (justusschock) [inline rising/transforms/channel.py:2]: imports of built-in packages should come before any other imports

File: rising/transforms/compose.py [hunk 2]
```diff
@@ -227,3 +234,54 @@ def forward(self, *seq_like, **map_like) -> Union[Sequence, Mapping]:
...
+            raise ValueError("If weights are porvided, every transform needs a weight. "
+                             f"Found {len(weights)} weights and {len(transforms)} transforms")
+        if weights is None:
+            self.weights = torch.tensor(
+                [1 / len(self.transforms)] * len(self.transforms))
+        else:
+            if not isinstance(weights, torch.Tensor):
+                self.weights = torch.tensor(weights)
+            else:
+                self.weights = weights
```
Comment (justusschock) [inline rising/transforms/compose.py:278]: can we transform the weights parameter and assign it all in one line after checking the length of weights is the same as the number of transforms?

File: rising/transforms/compose.py [hunk 3]
```diff
@@ -227,3 +234,54 @@ def forward(self, *seq_like, **map_like) -> Union[Sequence, Mapping]:
...
+        """
+        Args:
+            *transforms: transforms to choose from
+            weights: additional weights for transforms
+            p: probability that one transform i applied
+            transform_call: function which determines how transforms are
+                called. By default Mappings and Sequences are unpacked
+                during the transform.
+
+        References:
```
Comment (justusschock) [inline rising/transforms/compose.py:257]: Why do we need this reference? It's not like torchio invented this kind of transforms and as far as i can see, we did not copy their code?

File: rising/transforms/format.py [hunk 4]
```diff
@@ -108,3 +108,21 @@ def __init__(self, keys: Union[Callable, Sequence], return_popped: bool = False)
 
     def forward(self, **data) -> Union[dict, Tuple[dict, dict]]:
         return filter_keys(data=data, keys=self.keys, return_popped=self.return_popped)
+
+
+class RenameKeys(AbstractTransform):
```
Comment (justusschock) [inline rising/transforms/format.py:113]: For what is this trafo needed? For me it shouldn't matter which keys you specify.

File: rising/transforms/functional/channel.py [hunk 5]
```diff
@@ -6,25 +6,34 @@
 __all__ = ["one_hot_batch"]
 
 
-def one_hot_batch(target: torch.Tensor, num_classes: Optional[int] = None) -> torch.Tensor:
+def one_hot_batch(target: torch.Tensor,
+                  num_classes: Optional[int] = None,
+                  dtype: Optional[torch.dtype] = None) -> torch.Tensor:
```
Comment (justusschock) [inline rising/transforms/functional/channel.py:11]: If we add support for passing a specific dtype here, we also need to add support for passing devices, I think. Either both or nothing :D

File: rising/transforms/format.py (hunk 4 above)
Comment (mibaumgartner) [reply rising/transforms/format.py:113]: It is needed for external frameworks in the future. Generally speaking it is not necessary for us but might be a good transform to adjust rhe keys between a fixed dataset and already existing pipelines

File: rising/transforms/compose.py (hunk 3 above)
Comment (mibaumgartner) [reply rising/transforms/compose.py:257]: The code was inspired by their implementation and it was the first i saw something like this there. So I added just in case. We can also remove it if you want

File: rising/transforms/functional/channel.py (hunk 5 above)
Comment (mibaumgartner) [reply rising/transforms/functional/channel.py:11]: I added the dtype because the output is always of type long. When we think of segmentation pipelines the onehot encoding should be of type float for the rest of the transformations. If you prefer, we can also add the device here

File: rising/transforms/compose.py (hunk 3 above)
Comment (justusschock) [reply rising/transforms/compose.py:257]: I'd remove it. I've seen something similar in albumentations and others... I thing we either need to refer to all of them or None (And I'd prefer None in that case)
